Updated Script to 8 PAX 12PAX and 19pax

CaptainRafael opened the pull request:
This update adapts the passenger demand dynamically based on the aircraft selected. 
Whether operating an 8-seat, 12-seat, or 19-seat configuration, the script automatically adjusts the number of passengers to match the aircraft’s capacity, ensuring realistic loading and balanced mission generations .

This also adds a premium taking in count current rates.

CaptainRafael committed:
Updated Script to 8 PAX

CaptainRafael committed:
Updated Scripts for 8 12 and 19 pax seaters

CaptainRafael commented:
@skslater

ollie commented on corporate_upto19 .jsonplugin:
You got a typo in the filename. 🙃

CaptainRafael replied:
yeah just pushed yours suggestions and the same issue on the 12 json

Cheers

ollie replied:
Look like I messed up the white spaces, but at least it should parse now. Simon will prolly do something with it, just spotted this right away.

ollie commented on corporate_upto8.jsonplugin:
```diff
@@ -1,13 +1,15 @@
+    "maximumDistance": 1000,
     "passengerClass": ["first"]
+    "customMetadata": {
+        "multiplier": 1.7,}
```
```suggestion
        "multiplier": 1.7
      }
```

ollie commented on corporate_upto19 .jsonplugin:
```diff
@@ -0,0 +1,15 @@
+    "maximumDistance": 2000,
+    "passengerClass": ["first"]
+    "customMetadata": {
+        "multiplier": 1.7,}
```
```suggestion
        "multiplier": 1.7
      }
```

CaptainRafael committed:
Update corporate_upto19 .jsonplugin

Co-authored-by: Oldřich Vetešník <olda@fakturoid.cz>

CaptainRafael committed:
Update corporate_upto8.jsonplugin

Co-authored-by: Oldřich Vetešník <olda@fakturoid.cz>

CaptainRafael committed:
Update corporate_upto12.jsonplugin

fixed tipo

CaptainRafael committed:
Update corporate_upto19 .jsonplugin

This adds a bit more of pay bump my increasing minimum pax  
as at the moment the budget is per pax not the full plane

CaptainRafael committed:
Update corporate_upto12.jsonplugin

Adds a small bump in minimum pax for the pay , as they pay per pax seat and not for full aircraft

CaptainRafael committed:
Update corporate_upto8.jsonplugin

Adds a small bump in minimum pax for the pay , as they pay per pax seat and not for full aircraft

skslater commented:
@CaptainRafael I'm not sure I see the point of having multiple plugins here. The idea of plugins is that each one is a "type" of passenger with different requirements or expectations, there's nothing to be gained having multiple corporate types.


When it talks about maximum and minimum passenger counts that's only when generating a single group, there's nothing to stop larger aircraft loading 50 groups and a smaller aircraft only loading one group.


All it's really doing is forcing large groups to be generated. For example your `upto12` plugin is essentially saying these corporate passengers always turn up at the airport with at least 7 friends in their group, never less.

Plugin design is about looking at it from the viewpoint of a world builder, not a business owner or a pilot.

CaptainRafael commented:
I see your point about plugin philosophy being about types of passengers rather than business models, but the reason I went for multiple plugins (like the up-to-12 one) is more practical than cosmetic.

Right now, a corporate flight can easily end up being a 3hour sector with just 1 pax on board  and the payout barely covers the fuel. 
That doesn’t reflect how these flights usually work in reality, nor does it make them sustainable inside the sim. 
By shaping group sizes (7–12, 13–19, etc.), the plugin nudges the generator to produce missions that are closer to real corporate/charter usage and actually viable for operators.

So it’s less about saying “corporates always fly in packs of 7+” and more about using the group logic to balance economy and immersion. 
Different plugins just let me fine-tune that balance for various aircraft sizes.

the intent is to make corporate ops more realistic and sustainable in the sim, rather than purely chasing variety of “passenger types.”

skslater commented:
That's kinda my point though, plugins work at the group level, so a small group of related people, like a family or a group of friends, or in this case maybe a couple of business people travelling to a convention or a customer visit. It's not realistic to say that a solo business person can't travel further than 500 miles or that corporate fliers never book a trip over 1500 miles unless their group contains at least 13 people.

You're manipulating demand to fill aircraft but that's the opposite of how the platform works.The platform is passenger driven, not company driven. If you've got one lone corporate passenger that wants to fly 2000 miles it's your choice if you take them or not based on the pay, but that doesn't change their requirement to get from A to B. In the real world  you could say I'm not doing a flight of that distance with less than 20 passengers on board, and it's your job to fill it, but you can't tell an individual that he has to bring 19 colleagues if he wants to fly corporate long distance.

Or to put it another way...

"United Airlines, I'd like to book a ticket from New York to London for myself and my family of three"

"Of course, it's a big plane and a long flight though, so we only accept groups of at least 30"

CaptainRafael commented:
This has been discussed over discord , but FSC is designed for airlines , not corporate companies , the fact of increasing the minimum pax increases the minimum payout for corporate jets as they are not profitable at the moment , 

The system is designed for a solo travel to pay for the seat , IRL the solo will pay for the full aircraft , hence why this plugin was created to solve a bit of the issue for corporate / bizjets flights

skslater commented:
I don't disagree, but the demand engine isn't the place to resolve those kinds of issues. If something isn't profitable it has to be fixed in the platform economy side rather than generating unrealistic demand as a workaround. The demand engine is meant to create a steady flow of mixed passengers with differing requirements that suit different styles of play. Generating groups of 13-19 people that pay 1.7 times the normal cost just creates an imbalance for one type of airline or company versus others. The demand engine is separate from the platform partly for that reason, it's purely there to feed human travel demand into the system, it doesn't participate in the economy, has no concept of aircraft capabilities or capacities and it doesn't have any bias towards playstyle.

I do agree that maybe corporate passengers should pay a premium if that demand group is supposed to emulate those travelling on private jets, but that's a decision for Ekion and TheBlackSheep on the economy side. The use of multipliers has to be done in the context of the wider economy or we'll end up with hyperinflation.
